### ISSUE TITLE
Make partial parameters accessable in the PHP section of the partial template

### ISSUE THREAD

**Issue by PMK** [octobercms/october#1235]
Hi,

Along with other developers I'm unable to access the parameters of a partial. These parameters are not accessible inside the PHP section of the template of the partial, but are in the HTML.

```
{% partial "my-partial-name" paramName="value" %}
```
- https://octobercms.com/forum/post/accessing-partial-variables-in-onstart
- https://octobercms.com/forum/post/acces-partial-parameter-in-code-tab

In class Cms\Classes\Controller.php in function renderPartial() I do see the parameters with the values, including the name of the partial.

If I make an "else" state on line number 825, and putting the partial stuff inside the $this->vars array, then I can access it.

``` PHP
if (($partial = Partial::loadCached($this->theme, $name)) === null) {
    // ... existing code here ...

} else {
    // Obviously this is not correct...
    $this->vars[$name] = $parameters;
}
```

I think this should be done inside the Controller class via a getter, but if you paste this code here below inside your partial template in the PHP part, it will fix the problem.

``` PHP
$controller = $this->getController();
$partialParams = $controller->vars["your-partial-name-here"];
// var_dump($partialParams);
```

If you have access to the parameters in the PHP section, you can load your own class to enrich the partial with extra functionality. I am not sure if this is a bug or a "new feature". I did assume that this was possible because I was able to use the parameters in the HTML section.

**Comment by nsrosenqvist** [comment]
+1

**Comment by triasrahman** [comment]
+1

**Comment by estudiogenius** [comment]
+1

**Comment by vebam** [comment]
+1

**Comment by merodeador** [comment]
+1

**Comment by danilobrinu** [comment]
+1

**Comment by fresswolf** [comment]
+1

**Comment by snipiba** [comment]
+1

**Comment by Asinox** [comment]
And... any news about this?

**Comment by Beaudinn** [comment]
+1

**Comment by cracki** [comment]
+1

**Comment by Kilgaloon** [comment]
I worked a little bit on this issue and enabled parameters passed to partials to be available through method ``getParam( $param )`` and to get all parameters with ``getParams()``

**Comment by daftspunk** [comment]
Being looked at as part of #2517

**Comment by daftspunk** [comment]
This was fixed in Feb as part of ec03b7b2f85cbe89082eda9fe7e457534bdc522d
